community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -754,16 +772,25 @@ public void close( Consumer<PageCursor> headerWriter ) throws IOException
     private void close( Header.Writer headerWriter ) throws IOException
     {
         writer.close();
-
+        boolean didCheckpoint = false;
+        boolean anyWrites = false;
         try
         {
-            // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
-            // no new checkpoint will be created.
-            checkpoint( IOLimiter.unlimited(), headerWriter );
+            if ( health.isHealthy() )
+            {
+                // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
+                // no new checkpoint will be created.
+                anyWrites = checkpoint( IOLimiter.unlimited(), headerWriter );
+                didCheckpoint = true;
+            }
         }
         finally
         {
             pagedFile.close();
+            String checkpointMessage = didCheckpoint ?
```

Comment:
See above. We can't really know what to call this thing here. `index`, meh, for different uses this is a different thing, e.g. in label scan store this is a `label index`, for a schema index this is a `schema index for label L and property P`, not just `index`. This has to be a monitor to be done properly.